packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx

```diff
@@ -0,0 +1,175 @@
+      throw new Error('label text is required and must be string');
+  }
+  /**
+   * Define if the label is shown
+   *
+   * Allowed values:
+   * - true
+   * - false
+   *
+   * Default: true
+   */
+  @Prop() showLabel: boolean = true;
+  @Watch('showLabel')
+  validateShowLabel(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('show label must be boolean');
+  }
+  /**
+   * Define if an input is required.
+   *
+   * Allowed values:
+   * - true
+   * - false
+   *
+   * Default: false
+   */
+  @Prop() required: boolean = false;
+  @Watch('required')
+  validateRequired(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('required: must be boolean');
+  }
+  /**
+   *
+   * Define text input's size
+   *
+   * Allowed values:
+   *   - medium
+   *   - large
+   *
+   * Default: medium
+   */
+  @Prop() size: PlmgTextInputSize = 'medium';
+  @Watch('size')
+  validateSize(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('size: required');
+    if (!isPlmgTextInputSize(newValue))
+      throw new Error('size: must be a valid value');
+  }
+  /**
+   * Define tip
+   *
+   * Allowed value: any string
+   *
+   * Displays a tip message
+   */
+  @Prop() tip: string;
+  @Watch('tip')
+  validateTipText(newValue: string) {
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('tip text must be a string');
+  }
+
+  private handleInputChange(ev) {
+    this.value = ev.target.value;
+    this.valueUpdated.emit({ value: this.value });
+  }
+  /**
+   * Event emitted when value changed
+   */
+  @Event() valueUpdated: EventEmitter;
+  /**
+   * Life Cycle Methods & Event Listeners
+   */
+
+  render() {
+    const inputClasses = {
+      [this.size]: true,
+      ['error']: !!this.errorMessage,
+    };
+
+    const tipClasses = {
+      'plmg-text-input-tip': true,
+      [this.size]: true,
+    };
+
+    const labelClasses = {
+      [this.size]: true,
+      ['visually-hidden']: !this.showLabel,
+      'plmg-text-input-label': true,
+    };
+
+    return (
+      <div class={'plmg-text-input-wrapper'}>
+        <label class={labelClasses} htmlFor={this.labelToId()}>
+          {this.label}
+          {this.showLabel && this.required && <span class={'required'}>*</span>}
+        </label>
+        <div class={'plmg-text-input-field-wrapper'} tabIndex={0}>
+          <input
+            class={inputClasses}
+            id={this.labelToId()}
+            name={this.label}
+            required={this.required}
+            type={'text'}
+            value={this.value}
+            onInput={(ev) => this.handleInputChange(ev)}
+          />
+        </div>
+        {this.tip && <span class={tipClasses}>{this.tip}</span>}
+        {this.errorMessage && (
+          <plmg-error-message
+            size={this.size}
+            style={{
+              marginTop: tokens.plmgSpacingX05,
+            }}
```

Comment:
Why do an inline style rather than define a class?

Reply:
I thought it didn't work. Tried it again. It does now. Changed.